src/pages/SearchPage.js

```diff
@@ -170,31 +170,25 @@ class SearchPage extends Component {
             this.state.searchValue,
         );
         return (
-            <ScreenWrapper>
-                {({didScreenTransitionEnd}) => (
-                    <>
-                        <HeaderWithCloseButton
-                            title={this.props.translate('common.search')}
-                            onCloseButtonPress={() => Navigation.dismissModal(true)}
-                        />
-                        <View style={[styles.flex1, styles.w100, styles.pRelative]}>
-                            <FullScreenLoadingIndicator visible={!didScreenTransitionEnd} />
-                            {didScreenTransitionEnd && (
-                            <OptionsSelector
-                                sections={sections}
-                                value={this.state.searchValue}
-                                onSelectRow={this.selectReport}
-                                onChangeText={this.onChangeText}
-                                headerMessage={headerMessage}
-                                hideSectionHeaders
-                                hideAdditionalOptionStates
-                                showTitleTooltip
-                            />
-                            )}
-                        </View>
-                        <KeyboardSpacer />
-                    </>
-                )}
+            <ScreenWrapper onTransitionEnd={() => this.setState({isLoading: false})}>
```

Comment:
Does this assume loading will be done by the end of the transition?

Reply:
As of now (even without this PR), we show loading only for the transition period.  We could refactor it when we plan to use it for other actions like asynchronous data fetch, etc.

Reply:
Ok, so we are showing loading **because** the transition is happening?

Reply:
Yes @Luke9389 , see here.
In our current app, the following components where we showing loading for transition alone
 1: Search Page
Search Component,  until `didScreenTransitionEnd` we show `FullScreenLoadingIndicator` & hide the OptionsSelector. https://github.com/Expensify/App/blob/193ebc91dc1ecfd77fd5cbf0ee313afb32fc5dc7/src/pages/SearchPage.js#L174-L195

 2: New chat page
https://github.com/Expensify/App/blob/193ebc91dc1ecfd77fd5cbf0ee313afb32fc5dc7/src/pages/NewChatPage.js#L132-L144

 3: New group page
https://github.com/Expensify/App/blob/193ebc91dc1ecfd77fd5cbf0ee313afb32fc5dc7/src/pages/NewGroupPage.js#L195-L209

4: IOU Modal
https://github.com/Expensify/App/blob/193ebc91dc1ecfd77fd5cbf0ee313afb32fc5dc7/src/pages/iou/IOUModal.js#L322-L327